awpy/types.py

```diff
@@ -680,66 +687,124 @@ class RoundStatistics(TypedDict):
 
 
 @dataclass
-class BFSTileData(TypedDict):
+class BFSTileData:
     """Dataclass containing data for tiles during bfs algorithm.
 
     Holds information for tile_id for tile, current map control
     value, and steps remaining for bfs algorithm
     """
 
-    tile_id: TileId
-    map_control_value: float
-    steps_left: int
+    def __init__(
+        self,
+        *,
+        tile_id: TileId,
+        map_control_value: float,
+        steps_left: int,
+    ) -> None:
+        """Initialze BFSTileData class instance.
+
+        Args:
+            tile_id (TileId): TileId for tile
+            map_control_value (float): Map control value for tile
+            steps_left (int): Number of steps remaining for BFS
+                algorithm
+        """
+        self.tile_id = tile_id
+        self.map_control_value = map_control_value
+        self.steps_left = steps_left
 
 
 @dataclass
-class TeamMetadata(TypedDict):
+class TeamMetadata:
     """Dataclass containing metadata for one team.
 
     Holds information for aliver player locations. Can include
     more metadata (utility, bomb location, etc.) in the future
     """
 
-    alive_player_locations: list[list[float]] | list
+    def __init__(
+        self,
+        *,
+        alive_player_locations: list[list[float]],
+    ) -> None:
+        """Initialze TeamMetadata class instance.
+
+        Args:
+            alive_player_locations (list): List of player locations
+        """
+        self.alive_player_locations = alive_player_locations
 
 
 @dataclass
-class FrameTeamMetadata(TypedDict):
+class FrameTeamMetadata:
     """Dataclass with metadata on both teams in frame.
 
     Return type for awpy.analytics.map_control.extract_teams_metadata.
     Holds parsed metadata object (TeamMetadata) for both teams
     """
 
-    t: TeamMetadata
-    ct: TeamMetadata
+    def __init__(
+        self,
+        *,
+        t: TeamMetadata,
+        ct: TeamMetadata,
+    ) -> None:
+        """Initialze TeamMetadata class instance.
+
+        Args:
+            t (TeamMetadata): Object describing t side
+            ct (TeamMetadata): Object describing ct side
+        """
+        self.t = t
+        self.ct = ct
 
 
 @dataclass
-class FrameMapControlValues(TypedDict):
+class FrameMapControlValues:
     """Dataclass with map control values for both teams in frame.
 
     Return type for awpy.analytics.map_control.calc_map_control.
     Holds TeamMapControlValues for each team for a certain frame.
     """
 
-    t: TeamMapControlValues
-    ct: TeamMapControlValues
+    def __init__(
+        self,
+        *,
+        t: TeamMapControlValues,
+        ct: TeamMapControlValues,
+    ) -> None:
+        """Initialze FrameMapControlValues class instance.
+
+        Args:
+            t (TeamMapControlValues): Map contol values for t side
+            ct (TeamMapControlValues): Map contol values for ct side
+        """
+        self.t = t
+        self.ct = ct
```

Comment:
With data classes you can ditch the init function. The decorator handles that for you.

See here: https://github.com/pnxenopoulos/awpy/blob/1f32f001937ca203db0f5f2d46184c033a2a81e1/awpy/types.py#L15

Reply:
Oh I see thank you for letting me know! Addressed in next commit to branch.